cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1648,6 +1661,60 @@ String zkConnectionString(int connectToReplicas)  {
                     });
         }
 
+        Future<ReconciliationState> kafkaBrokerDynamicConfiguration() {
+            KafkaBrokerConfigurationHelper kbch = new KafkaBrokerConfigurationHelper(kafkaSetOperations.getAdminClientProvider(), secretOperations);
+            return kafkaSetOperations.getAsync(namespace, KafkaCluster.kafkaClusterName(name))
+                .compose(sts -> {
+                    if (sts == null) {
+                        return Future.succeededFuture();
+                    }
+                    int replicas = kafkaCluster.getReplicas();
+                    List<Future> configFutures = new ArrayList<>(replicas);
+
+                    for (int podId = 0; podId < replicas; podId++) {
+                        int finalPodId = podId;
+                        Future<Admin> acFut = kbch.adminClient(sts, podId);
+                        configFutures.add(
+                                acFut.compose(ac -> {
+                                    if (ac == null) {
+                                        kafkaPodsUpdatedDynamically.put(finalPodId, null);
+                                        log.warn("Could not create admin client.");
+                                        return Future.succeededFuture();
+                                    } else {
+                                        Future<Map<ConfigResource, Config>> futCurrent = kbch.getCurrentConfig(finalPodId, ac);
+                                        Future<ConfigMap> futDesired = configMapOperations.getAsync(namespace, KafkaCluster.metricAndLogConfigsName(name));
```

Comment:
Probably related to one of my other comments. This looks like wasteful HTTP call to the API server. We have generated this ConfgiMap few milliseconds ago. You should either store the desired configuration in `getKafkaAncialiaryCm` (as a field in `ReconciliationState` for example) or you should just generate it freshly.